torchao/quantization/quant_api.py

```diff
@@ -1645,25 +1690,16 @@ def _float8_dynamic_activation_float8_weight_quantize_tensor(weight, config):
             "PerRow quantization only works for bfloat16 precision input weight"
         )
 
-    block_size = get_block_size(weight.shape[-2:], weight_granularity)
-    if weight.dim() == 3:
-        block_size = tuple([1] + list(block_size))
-    quantized_weight = to_affine_quantized_floatx(
-        input_float=weight,
-        block_size=block_size,
-        target_dtype=weight_dtype,
-        scale_dtype=torch.float32,
-        _layout=Float8Layout(mm_config=mm_config),
+    quantized_weight = Float8Tensor.from_float(
+        weight,
+        weight_dtype,
+        weight_granularity,
     )
-
-    input_quant_func = _input_activation_quant_func_fp8
-    input_quant_kwargs = {
-        "activation_granularity": activation_granularity,
-        "activation_dtype": activation_dtype,
-    }
-
-    quantized_weight = to_linear_activation_quantized(
-        quantized_weight, input_quant_func, quant_kwargs=input_quant_kwargs
+    act_quant_config = Float8TensorConfig(
```

Comment:
I think the user should specify this, it's confusing to have some kinds of configs specified by user and some kinds specified in glue code.

i.e. can we just remove `Float8TensorConfig` and inline the logic?

Reply:
we can expose this to the workflow config as a user facing arg I think, but another purpose for this config is that `DynamicActivationQuantizationWrapper` can be used for different types of activation quantization, instead of we passing around the input_quant_func and kwargs as we do before, what do you have in mind for how to work with `DynamicActivationQuantizationWrapper` when we inline this config?